tools/scylla-nodetool.cc

```diff
@@ -81,9 +139,27 @@ const std::vector<operation_option> operation_options{
 };
 
 const std::vector<app_template::positional_option> operation_positional_options{
+    {"keyspace", bpo::value<sstring>(), "keyspace to process", 1},
+    {"tables", bpo::value<std::vector<sstring>>(), "tables to process", -1},
```

Comment:
We'll need to support all the existing documented options for each new command. Otherwise we'll quickly lose track of what's ported and what's not.

Does bpo support single-dash long options? e.g. `-pwf`.

For options that no longer make sense (`-pwf` again), we should accept them, but ignore or reject depending on what makes sense, with an explanatory message.

Reply:
`nodetool compact` has just these two positional arguments. `-pwf` and friends are global options, they are available with all commands. Those are declared in a separate vector, but so far I have only ported `--host` and `--port`. And here we already hit the first hickup. `nodetool` uses `-h` for the short form of `--host`. Its like the people who designed these C* cmd line apps, never saw a cmd line app before.

Reply:
> `nodetool compact` has just these two positional arguments. `-pwf` and friends are global options, they are available with all commands. 

The documentation mentions `--partition <partition_key> - String representation of the partition key.`.

> Those are declared in a separate vector, but so far I have only ported `--host` and `--port`. And here we already hit the first hickup. `nodetool` uses `-h` for the short form of `--host`. Its like the people who designed these C* cmd line apps, never saw a cmd line app before.

It comes from a different lineage. If the style clashes too much with bpo, we should hand-code rather than fight with it.

Reply:
> The documentation mentions --partition <partition_key> - String representation of the partition key..

Right, I forgot about that one. I will make sure to port all command line args, even unused ones.

Reply:
> It comes from a different lineage. If the style clashes too much with bpo, we should hand-code rather than fight with it.

Yes, I will probably pre-process the command line to remove anything that might confuse bpo.

Reply:
Done in v2.